### PR TITLE
Added MVC for user

### PR DESCRIPTION
**Pull request by shaneponcik**
and added change to dbsetup to allow for .env(MONGODB_CONNECT_STRING)

### CODE REVIEW

File: controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const User = require("../models/user.model");
+
+exports.create = function(req, res){
+    let user = new User(
+        {
+            username: req.body.username,
+            password: req.body.password
```
Comment (kevo1ution) [inline controllers/user.controller.js]: I'm not to sure if we will have a password field. We can leave it for now, but we are doing authentication with firebase, so we might need a firebase id or something whenever a user is getting created. We'll have to make a poc for that entire user story (make account on frontend -> firebase -> nodejs service).

File: controllers/user.controller.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const User = require("../models/user.model");
+
+exports.create = function(req, res){
+    let user = new User(
```
Comment (kevo1ution) [inline controllers/user.controller.js]: ```const user``` would be better here because user variable won't be changed

File: .env [hunk 3]
```diff
@@ -1,4 +1,5 @@
-MONGODB_USERNAME=user
-MONGODB_PASSWORD=password
-MONGODB_HOST=localhost:27017
-MONGODB_DBNAME=test
+MONGODB_USERNAME=shane
+MONGODB_PASSWORD=shane
+MONGODB_HOST=cluster0-j8cus.mongodb.net
+MONGODB_DBNAME=Cluster0
+MONGODB_CONNECT_STRING = mongodb+srv://shane:shane@cluster0-j8cus.mongodb.net/test?retryWrites=true&w=majority
```
Comment (kevo1ution) [inline .env]: don't put spaces: ```field=value``` instead of ```field = value```

File: .env [hunk 4]
```diff
@@ -1,4 +1,5 @@
-MONGODB_USERNAME=user
-MONGODB_PASSWORD=password
-MONGODB_HOST=localhost:27017
-MONGODB_DBNAME=test
+MONGODB_USERNAME=shane
```
Comment (kevo1ution) [inline .env]: don't commit changes to .env unless you are adding a new field.

File: controllers/user.controller.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            username: req.body.username,
+            password: req.body.password
+        }
+    );
+
+    user.save(function (err) {
```
Comment (kevo1ution) [inline controllers/user.controller.js]: we will need to set up some express middleware to handle errors, so you don't need to explicitly say ```res.send("Error")```

Comment (kevo1ution) [reply controllers/user.controller.js]: But for now, this looks fine

File: controllers/user.controller.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        if (err)
+        {
+            res.send("Error");
+        }
+        else
+        {
+            res.send('User created successfully');
```
Comment (kevo1ution) [inline controllers/user.controller.js]: use ```res.status()``` to give the HTTP status code as well (this should be applied to all routes)

File: models/user.model.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+let UserSchema = new Schema({
+    username: {type: String, required: true, max: 100},
+    password: {type: String, required: true, max: 25}
+});
+
+module.exports = mongoose.model('User', UserSchema);
```
Comment (kevo1ution) [inline models/user.model.js]: add new line

File: controllers/user.controller.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+exports.delete = function(req, res){
+    User.findByIdAndDelete(req.params.id, (err) => {
+        if (err)
+            res.send(err);
+        else
+            res.send("User Deleted with ID:" + req.params.id);
+    })
+}
```
Comment (kevo1ution) [inline controllers/user.controller.js]: add new line

File: routes/user.route.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+router.post('/create', user_controller.create);
+
+router.get('/:id', user_controller.details);
+
+router.put('/:id/update', user_controller.update);
+
+router.delete('/:id/delete', user_controller.delete);
+
+module.exports = router;
```
Comment (kevo1ution) [inline routes/user.route.js]: new line

File: src/services/dbSetup.js [hunk 10]
```diff
@@ -16,10 +16,17 @@ function setupDB() {
     };
   }
 
-  return mongoose.connect(
-    `mongodb://${process.env.MONGODB_HOST}/${process.env.MONGODB_DBNAME}`,
-    options
-  );
+  if(process.env.MONGODB_CONNECT_STRING)
+  {
+    return mongoose.connect(process.env.MONGODB_CONNECT_STRING);
```
Comment (kevo1ution) [inline src/services/dbSetup.js]: why not add options as an argument as well?

**Comment by kevo1ution** [comment]
Also the routes and models folder should be under `src`

File: controllers/user.controller.js (hunk 2 above)
Comment (shaneponcik) [reply controllers/user.controller.js]: cool

**Pull request merged**